utils_nlp/models/transformers/common.py

```diff
@@ -176,39 +149,47 @@ def fine_tune(
                 if gradient_accumulation_steps > 1:
                     loss = loss / gradient_accumulation_steps
 
-                if step % 10 == 0 and verbose:
-                    tqdm.write("Loss:{:.6f}".format(loss))
-
                 if fp16:
                     with amp.scale_loss(loss, optimizer) as scaled_loss:
                         scaled_loss.backward()
-                    torch.nn.utils.clip_grad_norm_(amp.master_params(optimizer), max_grad_norm)
+                    if clip_grad_norm:
+                        torch.nn.utils.clip_grad_norm_(amp.master_params(optimizer), max_grad_norm)
                 else:
                     loss.backward()
-                    torch.nn.utils.clip_grad_norm_(self.model.parameters(), max_grad_norm)
+                    if clip_grad_norm:
+                        torch.nn.utils.clip_grad_norm_(self.model.parameters(), max_grad_norm)
 
                 tr_loss += loss.item()
+
+                accum_loss += loss.item()
                 if (step + 1) % gradient_accumulation_steps == 0:
+                    global_step += 1
+                    if global_step % report_every == 0 and verbose:
+                        end = time.time()
+                        print(
+                            "loss: {0:.6f}, time: {1:f}, number of examples in current step: {2:.0f}, step {3:.0f} out of total {4:.0f}".format(
+                                accum_loss / report_every, end - start, len(batch), global_step, max_steps,
+                            )
+                        )
+                        accum_loss = 0
+                        start = end
+
                     optimizer.step()
-                    scheduler.step()
+                    if scheduler:
+                        scheduler.step()
                     self.model.zero_grad()
-                    global_step += 1
 
-                if max_steps > 0 and global_step > max_steps:
+                if global_step > max_steps:
                     epoch_iterator.close()
                     break
-            if max_steps > 0 and global_step > max_steps:
-                train_iterator.close()
-                break
 
         return global_step, tr_loss / global_step
 
     def predict(self, eval_dataloader, device, get_inputs, verbose=True):
         self.model.eval()
         for batch in tqdm(eval_dataloader, desc="Evaluating", disable=not verbose):
-            batch = tuple(t.to(device) for t in batch)
             with torch.no_grad():
-                inputs = get_inputs(batch, self.model_name, train_mode=False)
+                inputs = get_inputs(batch, device, self.model_name, train_mode=False)
```

Comment:
In the latest staging branch, there is move_batch_to_device in the fine-tune and predict functions. I am not sure why it's not showing us in the staging branch of your pull request.

Reply:
I removed this function and included the logic in get_inputs()

Reply:
I looked into your branch, the summarization work is not included: https://github.com/microsoft/nlp-recipes/tree/bleik/optim-patch/utils_nlp/models/transformers  Is this the place I should look at?

Reply:
right, I haven't included it yet. It needs to be modified to match the changes.